src/components/RulerTooltip/index.js

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  padding: PropTypes.number,
+  labelHeight: PropTypes.number.isRequired,
+  color: PropTypes.string.isRequired,
+  label: PropTypes.string.isRequired,
+  x: PropTypes.number.isRequired,
+  y: PropTypes.number.isRequired,
+  width: PropTypes.number.isRequired,
+};
+
+const defaultProps = {
+  padding: 10,
+};
+
+class RulerTooltip extends Component {
+  state = {
+    textWidth: 0,
+  };
+
+  componentDidMount() {
+    const textBBox = this.textRef.getBBox();
+    // eslint-disable-next-line react/no-did-mount-set-state
```

Comment:
Why are we able to ignore this?

Reply:
Is this a `setState` in did/will mount discussion?

Reply:
It seems the second render is not an issue [anymore](https://github.com/airbnb/javascript/issues/684#issuecomment-381674576), so airbnb want to turn this error off. The other problem we have I didn't think about before is that we can't really know in advance the text width has been already set in componentDidMount hook, but as far as it is synchronous operation we can run setState right after all the sync operations finish by wrapping it in an immediate setTimeout, i.e.
```
setTimeout(() =>
  this.setState({
    textWidth: textBBox.width,
  })
);
```